### DISCUSSION TITLE
Correct input structure description for spglib work ...

### DISCUSSION THREAD

**Discussion by husakm** [spglib/spglib#432] [Q&A] [answered]
I wan to know, what are the pre-requisites on the data to allow spglib identify correct lattice and space group ...
I believe it is required to generate all atoms in the unit cell (convert description to P1 group), delete all duplicity atoms and it should work ...  At lest it work on our side with following exception: rhombohedral and hexagonal systems ...
For this system we get incorrect resuslts. Table of example correct and spglib generated incorrect SG follows:
P3m1      P1
P-3c1      P-1
P6           P2
P62         P2_1
P63lmm  P2_1/m
Even the corrundum example form the demo code is identified as P-1 and not as R3c ....
Can anybody give me an example of rhombohedral /hexagonal data giving correct space group ?

**Comment by husakm** [comment]
After some test I see that the corundum works with following lattice description

```c
  double lattice[3][3] = {{4.8076344022756095, -2.4038172011378047, 0},
                          {0, 4.1635335244786962, 0},
                          {0, 0, 13.1172699198127543}};
```

How can I get form lattice parameters to this ? Standard crystallographic transformation gives me:

```c
  double lattice[3][3] = {{4.8076344022756095, 0, 0},
                          {-2.4038172011378047, 4.1635335244786962, 0},
                          {0, 0, 13.1172699198127543}};
```

Can anybody, please explain, what lattice description spglib use and how recalculate lattice parameters given in a,b,c, and alpha, beta,gamma to the correct required values ?

Michal

**Comment by LecrisUT** [comment]
Cannot assign, so CC @atztogo @lan496

**Comment by husakm** [comment]
I had find the issue:
a) On my input a parameter was different from c so 2 numbers in the matrix are wrong (corrected in my text)
b) Spglib use different convention than we used as standard (CASTEP code for QM calculations, we give cell vectors in rows, from Spglib documentation I see you use lattice vectors in columns ... 
I will check tomorrow all works after our code corrections ....
The convention we use:
https://www.tcm.phy.cam.ac.uk/castep/documentation/WebHelp/content/modules/castep/keywords/k_lattice_cart_castep.htm
I see the spglib convention now:
https://spglib.readthedocs.io/en/latest/definition.html

**Comment by atztogo** [comment]
I can't see what @husakm means. Please provide your question with a python script.

**Comment by husakm** [comment, marked as answer]
Problem solved on my side. Spglib use different lattice matrix description than our convention. See discusion on github for details ....

Michal Husak
UCT Prague ...
________________________________
